test/timezones/rounding.jl

```diff
@@ -0,0 +1,234 @@
+@test ceil(zdt, Dates.Hour) == ZonedDateTime(2015, 11, 1, 1, fixed)
+@test ceil(zdt, Dates.Minute(30)) == ZonedDateTime(2015, 11, 1, 1, fixed)
+@test round(zdt, Dates.Minute(30)) == ZonedDateTime(2015, 11, 1, 1, fixed)
+
+for tz in [winnipeg, st_johns]
+    zdt = ZonedDateTime(dt, tz)
+    next_hour = ZonedDateTime(DateTime(2015, 11, 1, 1), tz, 1)
+
+    @test ceil(zdt, Dates.Day) == ZonedDateTime(2015, 11, 2, tz)
+    @test ceil(zdt, Dates.Hour) == next_hour
+    @test ceil(zdt, Dates.Minute(30)) == next_hour
+    @test round(zdt, Dates.Minute(30)) == next_hour
+end
+
+# Test rounding forward and backward, during the ambiguous hour.
+
+dt = DateTime(2015, 11, 1, 1, 25)                   # During ambiguous hour
+
+zdt = ZonedDateTime(dt, fixed)
+@test floor(zdt, Dates.Day) == ZonedDateTime(2015, 11, 1, fixed)
+@test ceil(zdt, Dates.Day) == ZonedDateTime(2015, 11, 2, fixed)
+@test round(zdt, Dates.Day) == ZonedDateTime(2015, 11, 1, fixed)
+@test floor(zdt, Dates.Hour) == ZonedDateTime(2015, 11, 1, 1, fixed)
+@test ceil(zdt, Dates.Hour) == ZonedDateTime(2015, 11, 1, 2, fixed)
+@test round(zdt, Dates.Hour) == ZonedDateTime(2015, 11, 1, 1, fixed)
+@test floor(zdt, Dates.Minute(30)) == ZonedDateTime(2015, 11, 1, 1, fixed)
+@test ceil(zdt, Dates.Minute(30)) == ZonedDateTime(2015, 11, 1, 1, 30, fixed)
+@test round(zdt, Dates.Minute(30)) == ZonedDateTime(2015, 11, 1, 1, 30, fixed)
+
+for tz in [winnipeg, st_johns]
+    zdt = ZonedDateTime(dt, tz, 1)                  # First 1:25, before "falling back"
+    prev_hour = ZonedDateTime(DateTime(2015, 11, 1, 1), tz, 1)
+    between_hours = ZonedDateTime(DateTime(2015, 11, 1, 1, 30), tz, 1)
+    next_hour = ZonedDateTime(DateTime(2015, 11, 1, 1), tz, 2)
+    @test floor(zdt, Dates.Day) == ZonedDateTime(2015, 11, 1, tz)
+    @test ceil(zdt, Dates.Day) == ZonedDateTime(2015, 11, 2, tz)
+    @test round(zdt, Dates.Day) == ZonedDateTime(2015, 11, 1, tz)
+    @test floor(zdt, Dates.Hour) == prev_hour
+    @test ceil(zdt, Dates.Hour) == next_hour
+    @test round(zdt, Dates.Hour) == prev_hour
+    @test floor(zdt, Dates.Minute(30)) == prev_hour
+    @test ceil(zdt, Dates.Minute(30)) == between_hours
+    @test round(zdt, Dates.Minute(30)) == between_hours
+end
+
+###########################
+# ASIA/COLOMBO TRANSITION #
+###########################
+
+# Test rounding to ambiguous midnight, which (unfortunately) isn't handled well when
+# rounding to a DatePeriod resolution.
+
+zdt = ZonedDateTime(1996, 10, 25, 23, 55, colombo)  # 5 minutes before ambiguous half-hour
+@test floor(zdt, Dates.Day) == ZonedDateTime(1996, 10, 25, colombo)
+@test_throws AmbiguousTimeError ceil(zdt, Dates.Day)
+@test_throws AmbiguousTimeError round(zdt, Dates.Day)
+
+zdt = ZonedDateTime(1996, 10, 26, 0, 35, colombo)   # 5 minutes after ambiguous half-hour
+@test_throws AmbiguousTimeError floor(zdt, Dates.Day)
+@test_throws AmbiguousTimeError ceil(zdt, Dates.Day)    # floor is called during ceil
```

Comment:
I *think* I might even be able to fix most/all of the Colombo errors, but it will require a type check on the time zone. I'll take a look.